daemon/exec.go

```diff
@@ -102,12 +111,14 @@ func (d *Daemon) getExecConfig(name string) (*ExecConfig, error) {
 	// the user sees the same error now that they will after the
 	// 5 minute clean-up loop is run which erases old/dead execs.
 
-	if ExecConfig != nil && d.containers.Get(ExecConfig.Container.ID) != nil {
-
-		if !ExecConfig.Container.IsRunning() {
-			return nil, derr.ErrorCodeContainerNotRunning.WithArgs(ExecConfig.Container.ID)
+	if ec != nil && d.containers.Get(ec.Container.ID) != nil {
+		if !ec.Container.IsRunning() {
+			return nil, derr.ErrorCodeContainerNotRunning.WithArgs(ec.Container.ID)
```

Comment:
what do you think about combining these two errors in a single one. I'm pretty sure we have a lot of unnecessary duplication because we create an error per container state per feature.